client/src/components/food/FoodItem/index.js

```diff
@@ -36,8 +43,15 @@ function FoodItem({ label, calories }) {
         <span className={classes.calories}> {calories} Cal</span>
       </p>
       <div>
-        <EditIcon className={classes.icon} />
-        <HighlightOffTwoToneIcon className={classes.icon} color="primary" />
+        <EditIcon
+          className={classes.icon}
+          onClick={() => handleEditRequest(userFoodId, foodId)}
```

Comment:
I don't think `handleEditRequest` needs to be a prop that is passed down from this component's parent.  It is simple enough that the logic can just live inside this component (then the parent doesn't have to do quite so much).  So I'd recommend moving the `history.push` inside this file directly:
```
onClick={() => history.push(FoodQuantityPath, {
  categoryId: foodCategoryId,
  foodId,
  userFoodId,
  isEditFood: true,
}}
```
You'll just have to pass in `foodCategoryId` as a prop instead.  But overall, `FoodItemsSelected` and the interface between it, through `FoodItems` and into `FoodItem` would become simpler.